// @tosspayments/brandpay-types/esm/SDKBridge/message.js
import { MessageName as CancelMessageName } from './event/cancel';
import { MessageName as CommonParamsMessageName, } from './event/commonParams';
import { MessageName as CompleteMessageName } from './event/complete';
import { MessageName as CustomerTokenMessageName, } from './event/customerToken';
import { MessageName as DisposeBridgeMessageName, } from './event/disposeBridge';
import { MessageName as InvokedMethodMessageName, } from './event/invokedMethod';
import { MessageName as KTAlphaShopOCRScanMessageName, } from './event/kt-alpha-shop/ktAlphaShopOCRScan';
import { MessageName as MountMessageName } from './event/mount';
import { MessageName as GetShippingAddressWidgetEnabledMessageName, } from './event/payment-widget/getShippingAddressWidgetEnabled';
import { MessageName as SyncShippingAddressToBrandpayMessageName, } from './event/payment-widget/syncShippingAddressToBrandpay';
import { MessageName as PaymentMethodsMessageName, } from './event/paymentMethods';
import { MessageName as PaymentMethodTypeMessageName, } from './event/paymentMethodType';
import { MessageName as PaymentParamsMessageName, } from './event/paymentParams';
import { MessageName as PaymentResultMessageName, } from './event/paymentResult';
import { MessageName as RegisterResultMessageName, } from './event/registerResult';
import { MessageName as RegisterRoutePathMessageName, } from './event/RegisterRoutePath';
import { MessageName as RoutePathMessageName } from './event/routePath';
import { MessageName as SafeAreaMessageName } from './event/safeArea';
import { MessageName as SdkConfigMessageName } from './event/sdkConfig';
import { MessageName as ServerUrlMessageName } from './event/serverUrl';
import { MessageName as TermScopeMessageName } from './event/termScope';
import { MessageName as ViewportMessageName } from './event/viewport';
import { MessageName as WidgetMountMessageName, } from './event/widget/mount';
import { MessageName as OpenAddPaymentMethodMessageName, } from './event/widget/openAddPaymentMethod';
import { MessageName as OpenCashReceiptSettingMessageName, } from './event/widget/openCashReceiptSetting';
import { MessageName as OpenRegisterMessageName, } from './event/widget/openRegister';
import { MessageName as ResizeWidgetMessageName, } from './event/widget/resizeWidget';
import { MessageName as SyncStateMessageName, } from './event/widget/syncState';
import { MessageName as UpdateAmountMessageName, } from './event/widget/updateAmount';
import { MessageName as UpdateCashReceiptMessageName, } from './event/widget/updateCashReceipt';
import { MessageName as UpdatePaymentMethodMessageName, } from './event/widget/updatePaymentMethod';
import { MessageName as UpdateWidgetParamsMessageName, } from './event/widget/updateWidgetParams';
import { MessageName as WidgetParamsMessageName, } from './event/widget/widgetParams';
export const MessageType = {
    COMMON_PARAMS: CommonParamsMessageName,
    PAYMENT_PARAMS: PaymentParamsMessageName,
    METHOD_TYPE: PaymentMethodTypeMessageName,
    SAFE_AREA: SafeAreaMessageName,
    VIEWPORT: ViewportMessageName,
    PAYMENT_RESULT: PaymentResultMessageName,
    INVOKED_METHOD: InvokedMethodMessageName,
    ROUTE_PATH: RoutePathMessageName,
    CUSTOMER_TOKEN: CustomerTokenMessageName,
    PAYMENT_METHODS: PaymentMethodsMessageName,
    COMPLETE: CompleteMessageName,
    CANCEL: CancelMessageName,
    REGISTER_RESULT: RegisterResultMessageName,
    TERM_SCOPE: TermScopeMessageName,
    SERVER_URL: ServerUrlMessageName,
    DISPOSE_BRIDGE: DisposeBridgeMessageName,
    REGISTER_ROUTE_PATH: RegisterRoutePathMessageName,
    SDK_CONFIG: SdkConfigMessageName,
    KT_ALPHA_SHOP_OCR_SCAN: KTAlphaShopOCRScanMessageName,
    MOUNT: MountMessageName,
};
export const WidgetMessageType = {
    OPEN_CASH_RECEIPT_SETTING: OpenCashReceiptSettingMessageName,
    OPEN_ADD_PAYMENT_METHOD: OpenAddPaymentMethodMessageName,
    WIDGET_PARAMS: WidgetParamsMessageName,
    COMMON_PARAMS: CommonParamsMessageName,
    VIEWPORT: ViewportMessageName,
    SAFE_AREA: SafeAreaMessageName,
    INVOKED_METHOD: InvokedMethodMessageName,
    RESIZE_WIDGET: ResizeWidgetMessageName,
    UPDATE_PAYMENT_METHOD: UpdatePaymentMethodMessageName,
    SYNC_STATE: SyncStateMessageName,
    UPDATE_CASH_RECEIPT: UpdateCashReceiptMessageName,
    UPDATE_WIDGET_PARAMS: UpdateWidgetParamsMessageName,
    SERVER_URL: ServerUrlMessageName,
    UPDATE_AMOUNT: UpdateAmountMessageName,
    DISPOSE_BRIDGE: DisposeBridgeMessageName,
    OPEN_REGISTER: OpenRegisterMessageName,
    CUSTOMER_TOKEN: CustomerTokenMessageName,
    SDK_CONFIG: SdkConfigMessageName,
    MOUNT: WidgetMountMessageName,
};
export const PaymentWidgetMessageType = {
    SYNC_SHIPPING_ADDRESS_TO_BRANDPAY: SyncShippingAddressToBrandpayMessageName,
    GET_SHIPPING_ADDRESS_WIDGET_ENABLED: GetShippingAddressWidgetEnabledMessageName,
};
