// @tosspayments/brandpay-types/dist/apiClient/methods/account.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BankCode = exports.BankCodeInNumber = exports.BankCodeInEnglish = exports.BankCodeInKorean = void 0;
exports.BankCodeInKorean = {
    농협: `농협`,
    우리: `우리`,
    신한: `신한`,
    국민: `국민`,
    하나: `하나`,
    씨티: `씨티`,
    기업: `기업`,
    산업: `산업`,
    SC제일: `SC제일`,
    대구: `대구`,
    부산: `부산`,
    전북: `전북`,
    경남: `경남`,
    제주: `제주`,
    광주: `광주`,
    새마을: `새마을`,
    수협: `수협`,
    신협: `신협`,
    우체국: `우체국`,
    케이: `케이`,
    카카오: `카카오`,
    저축: `저축`,
    산림: `산림`,
    SBI: `SBI`,
    토스: `토스`,
};
exports.BankCodeInEnglish = {
    NONGHYEOP: 'NONGHYEOP',
    WOORI: 'WOORI',
    SHINHAN: 'SHINHAN',
    KOOKMIN: 'KOOKMIN',
    HANA: 'HANA',
    CITI: 'CITI',
    IBK: 'IBK',
    KDBBANK: 'KDBBANK',
    SC: 'SC',
    DAEGUBANK: 'DAEGUBANK',
    BUSANBANK: 'BUSANBANK',
    JEONBUKBANK: 'JEONBUKBANK',
    KYONGNAMBANK: 'KYONGNAMBANK',
    JEJUBANK: 'JEJUBANK',
    GWANGJUBANK: 'GWANGJUBANK',
    SAEMAUL: 'SAEMAUL',
    SUHYEOP: 'SUHYEOP',
    SHINHYEOP: 'SHINHYEOP',
    POST: 'POST',
    KBANK: 'KBANK',
    KAKAOBANK: 'KAKAOBANK',
    SAVINGBANK: 'SAVINGBANK',
    SANLIM: 'SANLIM',
    TOSSBANK: 'TOSSBANK',
};
exports.BankCodeInNumber = {
    ['39']: '경남',
    ['34']: '광주',
    ['06']: '국민',
    ['03']: '기업',
    ['11']: '농협',
    ['31']: '대구',
    ['32']: '부산',
    ['02']: '산업',
    ['45']: '새마을',
    ['64']: '산림',
    ['07']: '수협',
    ['88']: '신한',
    ['48']: '신협',
    ['27']: '씨티',
    ['20']: '우리',
    ['71']: '우체국',
    ['50']: '저축',
    ['37']: '전북',
    ['35']: '제주',
    ['90']: '카카오',
    ['89']: '케이',
    ['92']: '토스',
    ['81']: '하나',
    ['23']: 'SC제일',
};
exports.BankCode = Object.assign(Object.assign({}, exports.BankCodeInKorean), exports.BankCodeInEnglish);
